test/e2e/network_test.go

```diff
@@ -574,6 +575,30 @@ func (tc *testContext) getPodEvents(name string) ([]v1.Event, error) {
 	return podEvents, nil
 }
 
+// createLinuxCurlerJob creates a linux job to curl a specific endpoint. curl must be present in the container image.
+func (testCtx *testContext) createLinuxCurlerJob(jobSuffix, endpoint, portNum, portName string) (*batch.Job, error) {
```

Comment:
I suggest making the signature much simpler i.e. combine endpoint, portNum and portName into a URL and construct that at the call site. WDYT?